lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -318,18 +327,35 @@ private TopFieldCollector(FieldValueHitQueue<Entry> pq, int numHits,
       scoreMode = needsScores ? ScoreMode.COMPLETE : ScoreMode.COMPLETE_NO_SCORES;
       canSetMinScore = false;
     }
+    this.bottomValueChecker = bottomValueChecker;
   }
 
   @Override
   public ScoreMode scoreMode() {
     return scoreMode;
   }
 
+  protected boolean shouldUpdateMinScore() {
```

Comment:
Should we call it `betterGlobalScoreAvailable`, since the minimum score could still be left unchanged (if the threshold is not reached, for eg?) Another idea is to refactor `updateMinCompetitiveScore` to be split into two discrete methods, so that `shouldUpdateMinScore` can be extended to also update the score?

Reply:
+1 for renaming, I am not sure about the splitting of `updateMinCompetitiveScore` since `shouldUpdateMinScore` is a special thing that is only needed when we collect a non-competitive hits.